action.nix

```diff
@@ -84,33 +84,33 @@ with builtins; with lib; let
       NIXPKGS_ALLOW_UNFREE=1 nix-build --no-out-link \
          --argstr bundle "${bundlestr}" --argstr job "${job}" \
          --dry-run 2> err > out || (touch fail; true)
+      cat out err
+      if [ -e fail ]; then echo "Error: getting derivation failed"; exit 1; fi
     '';
   };
 
-  stepErrorReporting = {
-    name = "Error reporting";
-    run = ''
-        echo "out="; cat out
-        echo "err="; cat err
-    '';
-  };
-
-  stepFailureCheck = {
-    name = "Failure check";
-    run = "if [ -e fail ]; then exit 1; else exit 0; fi;";
-  };
-
   stepCheck = {
     name = "Checking presence of CI target for current job";
     id = "stepCheck";
-    run = "(echo -n status=; cat out err | grep \"built:\" | sed \"s/.*/built/\") >> $GITHUB_OUTPUT";
+    run = ''
+        if $(cat out err | grep -q "built:") ; then
+          echo "CI target needs actual building"
+          if $(cat out err | grep -q "derivations will be built:") ; then
+            echo "waiting a bit for derivations that should be in cache"
+            sleep 30
+          fi
+        else
+          echo "CI target already built"
+          echo "status=fetched" >> $GITHUB_OUTPUT
```

Comment:
The logic looks strange to me. The two conditionals seem to test pretty much the same thing, except that one is right after the other. Did you intend for the second one to actually test something else, or is it me that is reading wrong?

Reply:
Yes, everything lies in the plural. That being said, I didn't observe the issue anymore, so maybe it's no longer needed.

Reply:
Well, it can make some pull_request_target jobs (with cachix push disactivated for safety reasons) a bit slower, but indeed nothing dramatic.